odml/property.py

```diff
@@ -373,45 +373,73 @@ def clone(self):
             self_def = ''.join(map(str.strip, self.definition.split())).lower()
-            other_def = ''.join(map(str.strip, other.definition.split())).lower()
+            other_def = ''.join(map(str.strip, source.definition.split())).lower()
             if self_def != other_def:
-                raise ValueError("odml.Property.merge: src and dest definitions do not match!")
+                raise ValueError("odml.Property.merge: "
+                                 "src and dest definitions do not match!")
 
-        if self.uncertainty is not None and other.uncertainty is not None:
-            raise ValueError("odml.Property.merge: src and dest uncertainty both set and do not match!")
+        if (self.uncertainty is not None and source.uncertainty is not None and
+                self.uncertainty != source.uncertainty):
+            raise ValueError("odml.Property.merge: "
+                             "src and dest uncertainty both set and do not match!")
 
-        if self.reference is not None and other.reference is not None:
+        if self.reference is not None and source.reference is not None:
             self_ref = ''.join(map(str.strip, self.reference.lower().split()))
-            other_ref = ''.join(map(str.strip, other.reference.lower().split()))
+            other_ref = ''.join(map(str.strip, source.reference.lower().split()))
             if self_ref != other_ref:
-                raise ValueError("odml.Property.merge: src and dest references are in conflict!")
+                raise ValueError("odml.Property.merge: "
+                                 "src and dest references are in conflict!")
 
-        if self.value_origin is not None and other.value_origin is not None:
+        if self.value_origin is not None and source.value_origin is not None:
             self_ori = ''.join(map(str.strip, self.value_origin.lower().split()))
-            other_ori = ''.join(map(str.strip, other.value_origin.lower().split()))
+            other_ori = ''.join(map(str.strip, source.value_origin.lower().split()))
             if self_ori != other_ori:
-                raise ValueError("odml.Property.merge: src and dest value_origin are in conflict!")
+                raise ValueError("odml.Property.merge: "
+                                 "src and dest value_origin are in conflict!")
+
+    def merge(self, other, strict=True):
+        """
+        Merges the property 'other' into self, if possible. Information
+        will be synchronized. Method will raise a ValueError when the
+        information in this property and the passed property are in
+        conflict.
+
+        :param other: an odML Property.
+        :param strict: Bool value to indicate whether types should be implicitly converted
+               even when information may be lost. Default is True, i.e. no conversion,
+               and a ValueError will be raised if types do not match.
+        """
+        assert(isinstance(other, BaseProperty))
```

Comment:
Asserts aren't particularly nice outside of tests. The problem here is that if the types don't match, the user gets an AssertionError. It would be more appropriate and informative to check the type and manually raise a TypeError with a nicely crafted message.

Reply:
I just kept it as I found it, but I totally agree! Lets change it if everone's in.